### PR TITLE
feat: project page initial implementation

### PR DESCRIPTION
**Pull request by chetannn**

### CODE REVIEW

**Comment by chetannn** [comment]
image
This is working in progress 😀

**Comment by chetannn** [comment]
https://user-images.githubusercontent.com/22125423/216657426-eadd743b-e3bc-4ab9-9512-5fa53a950e73.mov

Implemented drag and drop environment variables and simple version of environment variable editor.

**Comment by chetannn** [comment]
Fixed delete issue and click to add manually along with pasting env variables.

https://user-images.githubusercontent.com/22125423/216797615-655ed268-4388-4bda-9277-cd96e6847da9.mov

**Comment by dahal** [comment]
Here are few feedbacks and decisions we came up with

- [ ] Input for env key/value form. Let's create an input group for value, and this input's values should be masked by default ***** and when you click on eye icon, it will revel the secrets.
- [ ] Lets change key/value input font to `font-mono`
- [ ] Fix container component's classes
- [ ] create icon button, lets stick with `rounded`
- [ ] Make key/value full width

File: hooks/useCopyToClipBoard.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+
+type CopiedValue = string | null;
+type CopyFn = (text: string) => Promise<boolean>;
+
+function useCopyToClipBoard(): [CopiedValue, CopyFn] {
```
Comment (dahal) [inline hooks/useCopyToClipBoard.ts:6]: This is awesome, thanks for creating a hook. I will replace my custom one on Homepage, Blog and Changelog click to copy thing.

File: pages/projects/[id]/index.tsx [hunk 2]
```diff
@@ -2,6 +2,9 @@ import { type GetServerSidePropsContext } from "next";
 import ProjectLayout from "@/layouts/Project";
 import { getServerSideSession } from "@/utils/session";
 import { Project } from "@prisma/client";
+import { HiOutlineCommandLine } from "react-icons/hi2";
```
Comment (dahal) [inline pages/projects/[id]/index.tsx:5]: Lets replace these icons with lucide.dev's. `main` does not have these icon libraries anymore.

File: pages/projects/[id]/index.tsx [hunk 3]
```diff
@@ -19,7 +22,38 @@ interface Props {
...
+            </p>
+          </div>
+        </div>
+      </div>
+
+      <div className="mt-8 w-full">
+        <div className="flex w-full items-center justify-between">
+          <Button outline small>
+            main
+          </Button>
```
Comment (dahal) [inline pages/projects/[id]/index.tsx:45]: Lets replace this with Popover with searchable select (similar to what we have on project picker, top navigation).

File: pages/projects/[id]/index.tsx (hunk 2 above)
Comment (chetannn) [reply pages/projects/[id]/index.tsx:5]: Sure.. Will do that.. Thanks for the feedback

**Pull request closed without merging**